db-developments/python/geocode_hpd_historical.py

```diff
@@ -0,0 +1,122 @@
+from multiprocessing import Pool, cpu_count
+from sqlalchemy import create_engine, text
+from geosupport import Geosupport, GeosupportError
+from utils import psql_insert_copy
+import pandas as pd
+import json
+import os
+
+g = Geosupport()
+
+
+def geocode(input):
+    # collect inputs
+    uid = str(input.pop("ogc_fid"))
+    hnum = input.pop("house_number")
+    sname = input.pop("street_name")
+    borough = input.pop("boro_short_name")
+
+    try:
+        geo = g["1B"](
+            street_name=sname,
+            house_number=hnum,
+            borough=borough,
+            mode="regular",
+        )
+        geo = parse_output(geo)
+        geo.update(
+            dict(
+                uid=uid,
+                mode="regular",
+                func="1B",
+                status="success",
+            )
+        )
+    except GeosupportError:
+        try:
+            geo = g["1B"](
+                street_name=sname,
+                house_number=hnum,
+                borough=borough,
+                mode="tpad",
+            )
+            geo = parse_output(geo)
+            geo.update(
+                dict(
+                    uid=uid,
+                    mode="tpad",
+                    func="1B",
+                    status="success",
+                )
+            )
+        except GeosupportError as e:
+            geo = parse_output(e.result)
+            geo.update(
+                uid=uid,
+                mode="tpad",
+                func="1B",
+                status="failure",
+            )
+    return geo
+
+
+def parse_output(geo):
+    return dict(
+        # Normalized address:
+        geo_sname=geo.get("First Street Name Normalized", ""),
+        geo_hnum=geo.get("House Number - Display Format", ""),
+        # Longitude and latitude of lot center
+        geo_latitude=geo.get("Latitude", ""),
+        geo_longitude=geo.get("Longitude", ""),
+        # Some sample administrative areas:
+        geo_bin=geo.get(
+            "Building Identification Number (BIN) of Input Address or NAP", ""
+        ),
+        geo_bbl=geo.get("BOROUGH BLOCK LOT (BBL)", {}).get(
+            "BOROUGH BLOCK LOT (BBL)",
+            "",
+        ),
+    )
+
+
+if __name__ == "__main__":
+    # connect to postgres db
+    engine = create_engine(os.environ["BUILD_ENGINE"])
+
+    # read in housing table
+    select_query = """
+            SELECT * 
+            FROM hpd_historical_units_by_building
+            WHERE reporting_construction_type = 'New Construction'
+            AND project_name <> 'CONFIDENTIAL';
+            """
+    with engine.begin() as conn:
+        df = pd.read_sql(
+            text(select_query),
+            conn,
+        )
+
+    # get the row number
```

Comment:
is that what this is doing?

Reply:
nope! lol this is also inaccurate in `geocode_hpd_hny.py` so this is copypasta